### ISSUE TITLE
Testing issues

### ISSUE THREAD

**Issue by jbarrett** [duckduckgo/community-platform#473]
So now Travis CI builds are passing, let's take a look at their performance:

![Passing - whoopee](http://jbrt.org/ddg/travis.png)

Getting close to a half-hour to finish some runs - we must be covering absolutely everything!

![Low test cover areas](http://jbrt.org/ddg/coverage.png)

Oh... ([full report](http://jbrt.org/ddg/ddgc-cover/coverage.html))

There are certain things which take time in Travis that we can't do much about, like installing dependencies, but the tests themselves take a fair amount of time and spend most of that time testing [just the](https://github.com/duckduckgo/community-platform/blob/e31fb9b/t/ddgc.t) [schema classes](https://github.com/duckduckgo/community-platform/blob/e31fb9b/lib/DDGCTest/Database.pm).

We should probably move to something like [`Test::WWW::Mechanize`](https://metacpan.org/pod/Test::WWW::Mechanize) and POST data to the various HTTP endpoints, then check if pages, notifications, moderation tasks etc. have been generated correctly over HTTP. This tests the application end-to-end rather than just testing the Model. We also don't need to test the same paths hundreds of times (testing the database = testing a dev deployment which makes hundreds of similar records).

We will also need to move away from SQLite completely soon. Some DDGC code requires/will require postgres. We can configure Travis for this:

http://docs.travis-ci.com/user/using-postgresql/

For testing in other environments, there will need to be a 'testing' user of some type with createdb permission (I guess).

Is there a JavaScript unit testing tool we could integrate which would be suitable for DDGC Ajax calls and IA Pages?

In summary:
- Tests need to take less time - aim for < 1 minute on a standard dev instance.
- Tests need to cover more.
- SQLite will be deprecated.
- It would be nice to have some sort of automated testing for our js.

@jdorweiler @MariagraziaAlastra @russellholt @moollaza 

Thanks!
